### DISCUSSION TITLE
How to load properties of a navigation property?

### DISCUSSION THREAD

**Discussion by benjamin-almarza** [OfficeDev/office-js#3207] [Q&A]
I don't have problems retrieving scalar properties, but I have linting errors when those properties belong to navigation property.

As you can see in my code the id is not directly in "selection" but one level deeper. 
```tsx
 const myFunction = async (context: Word.RequestContext): Promise<MyOutputType> => {
  const selection = context.document.getSelection();

  selection.load([
    "parentContentControlOrNullObject/id",
    "parentContentControlOrNullObject/parentContentControlOrNullObject/id",
  ]);

  await context.sync();

  const id = selection.parentContentControlOrNullObject.id;
  const parentId = selection.parentContentControlOrNullObject.parentContentControlOrNullObject.id;

 return { id,  parentId };
}
```

The warning look like this:
```
Calling load on the navigation property 'parentContentControlOrNullObject/id' slows down your add-in.eslintoffice-addins/no-navigational-load
```

And gives me a links to this [page](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model#scalar-and-navigation-properties)

What can I do to avoid those warning? What is the best and most efficient way to do this.

**Comment by timagixe** [comment]
@benjamin-almarza did you find out the answer?

**Reply by benjamin-almarza** [reply]
@timagixe , I didn't receive any answer that can fix this,  
I also posted the question here

https://learn.microsoft.com/en-us/answers/questions/1189264/in-office-addin-avoid-warnings-when-loading-nested

:p please let me know if you have a solution for this
